PetAdoptions/petlistadoptions-py/app.py

```diff
@@ -0,0 +1,235 @@
+    cuteness_rate: Optional[str] = None
+    petcolor: Optional[str] = None
+    petid: Optional[str] = None
+    pettype: Optional[str] = None
+    peturl: Optional[str] = None
+    price: Optional[str] = None
+
+class HealthResponse(BaseModel):
+    status: str
+
+# Create FastAPI app
+app = FastAPI(
+    title="Pet List Adoptions Service",
+    description="Service for listing pet adoptions with enrichment from pet search",
+    version="1.0.0"
+)
+
+class DatabaseConfig:
+    """Database configuration from AWS Secrets Manager"""
+    def __init__(self, engine: str, host: str, username: str, password: str, dbname: str, port: int):
+        self.engine = engine
+        self.host = host
+        self.username = username
+        self.password = password
+        self.dbname = dbname
+        self.port = port
+
+class PetAdoptionsService:
+    """Main service class following Python best practices"""
+    
+    def __init__(self):
+        self.pet_search_url = os.getenv("APP_PET_SEARCH_URL")
+        self.rds_secret_arn = os.getenv("APP_RDS_SECRET_ARN")
+        
+        # If not set via env vars, try to get from Parameter Store
+        if not self.pet_search_url or not self.rds_secret_arn:
+            self._fetch_from_parameter_store()
+    
+    def _fetch_from_parameter_store(self):
+        """Fetch configuration from AWS Parameter Store"""
+        try:
+            ssm = boto3.client('ssm')
+            response = ssm.get_parameters(
+                Names=[
+                    '/petstore/rdssecretarn',
+                    '/petstore/searchapiurl'
+                ]
+            )
+            
+            for param in response['Parameters']:
+                if param['Name'] == '/petstore/rdssecretarn':
+                    self.rds_secret_arn = param['Value']
+                elif param['Name'] == '/petstore/searchapiurl':
+                    self.pet_search_url = param['Value']
+                    
+        except Exception as e:
+            logger.error(f"Failed to fetch from Parameter Store: {e}")
+    
+    def _get_database_connection_string(self) -> str:
+        """Get database connection string from AWS Secrets Manager"""
+        try:
+            # Check if this is a local test setup
+            if self.rds_secret_arn == "local-secret":
+                # Read from local file for testing
+                with open("/app/local-secret.json", "r") as f:
+                    secret_data = json.loads(f.read())
+            else:
+                # Use AWS Secrets Manager
+                secrets = boto3.client('secretsmanager')
+                response = secrets.get_secret_value(SecretId=self.rds_secret_arn)
+                secret_data = json.loads(response['SecretString'])
+            
+            # Build connection string for psycopg2
+            connection_string = f"postgresql://{secret_data['username']}:{secret_data['password']}@{secret_data['host']}:{secret_data.get('port', 5432)}/{secret_data['dbname']}"
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Log Injection occurs when unsafe input is directly written to log files without proper sanitization. This can allow attackers to manipulate log entries, potentially leading to security issues like log forging or cross-site scripting. To prevent this, always sanitize user input before logging by removing or encoding newline characters, using string encoding functions, and leveraging built-in sanitization features of logging libraries when available. Learn more - https://cwe.mitre.org/data/definitions/117.html